### DISCUSSION TITLE
Similar UPDATE statements

### DISCUSSION THREAD

**Discussion by manticore-projects** [prestodb/sql#37] [Ideas]
Greetings and compliments.

I stumbled over the EBNF of the `UPDATE ...` statement: there are actually 2 different `UPDATE ...` statements, while only the `WHERE ...` clause is different.
Is this intended or would it not be better to refactor it into one single `UPDATE ...` statement with an abstract `WHERE ...` clause, which can be implement with 2 different forms?

I would suggest refactoring `CURRENT OF cursorname` into a `search_condition` and this unifying the statement.

![image](https://user-images.githubusercontent.com/18080123/210332289-18b97bf3-4822-4ee6-bee5-099fdcb35e60.png)

**Comment by kaikalur** [comment]
This is Oracle grammar looks like. Like I said before this grammar is actually extracted directly from the spec and I want to keep it that way for the spec part even though there are some redundancies it shouldn't have material impact on the performance.

**Reply by manticore-projects** [reply]
Ok, got it. Thanks!
